velox/exec/tests/utils/PlanBuilder.cpp

```diff
@@ -1430,11 +1432,22 @@ PlanBuilder& PlanBuilder::unnest(
     unnestFields.emplace_back(field(name));
   }
 
+  bool isLegacyUnnest = legacyUnnest.has_value() ? legacyUnnest.value() : true;
   std::vector<std::string> unnestNames;
   for (const auto& name : unnestColumns) {
     auto input = planNode_->outputType()->findChild(name);
     if (input->isArray()) {
-      unnestNames.push_back(name + "_e");
+      if (input->childAt(0)->isRow() && !isLegacyUnnest) {
+        // The array of rows is unnested into multiple columns, one for each
+        // child type of the row.
+        auto row = asRowType(input->childAt(0));
+        auto size = row->size();
+        for (auto i = 0; i < size; i++) {
+          unnestNames.push_back(row->nameOf(i) + "_e");
```

Comment:
Should we include 'name' as well? What if there are multiple columns of the same array(row) type?